group_vars/all/main.yml

```diff
@@ -17,6 +17,8 @@ wordpress_env_defaults:
   wp_siteurl: "${WP_HOME}/wp"
 
 site_env: "{{ wordpress_env_defaults | combine(item.value.env | default({}), vault_wordpress_sites[item.key].env) }}"
+site_hosts: "[{% for item in item.value.site_hosts %}'{{ item.canonical }}',{% for redirect in item.redirects | default([]) %}'{{ redirect }}',{% endfor %}{% endfor %}]"
+site_hosts_canonical: "{{ item.value.site_hosts | map(attribute='canonical') | list }}"
```

Comment:
Alternative:

```yaml
site_hosts_canonical: "{{ item.value.site_hosts | map(attribute='canonical') | list }}"
site_hosts_redirects: "{{ item.value.site_hosts | selectattr('redirects', 'defined') | map(attribute='redirects') | map('first') | list }}"
site_hosts: "{{ site_hosts_canonical | union(site_hosts_redirects) }}"
```

These have the advantage of making usage in other roles simpler. No need for looping or `default`.